### PR TITLE
Update the CurvedScalarWave system

### PR DESCRIPTION
**Pull request by prayush**
## Proposed changes

Several updates to the curved SW system to bring it up-to-speed with recent changes in code organization of non-conservative evolution systems:

 - Move time derivatives to their own header/source files
 - Update time derivative computer struct to work with `dg::Actions::ComputeTimeDerivative`
 - Move upwind correction computer struct to separate header/source files
 - Update tests accordingly

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [x] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Evolution/Systems/CurvedScalarWave/TimeDerivative.hpp
Comment (nilsdeppe) [inline src/Evolution/Systems/CurvedScalarWave/TimeDerivative.hpp]: I don't think you need the `return_tags` anymore. These are now known because we don't have to deal with the elliptic solver anymore

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
Please squash immediately

File: tests/Unit/Evolution/Systems/CurvedScalarWave/Test_DuDt.cpp
Comment (nilsvu) [inline tests/Unit/Evolution/Systems/CurvedScalarWave/Test_DuDt.cpp:96]: Please rename this file `Test_DuDt.cpp` -> `Test_TimeDerivative.cpp`

File: tests/Unit/Evolution/Systems/CurvedScalarWave/Test_UpwindFlux.cpp
Comment (nilsvu) [inline tests/Unit/Evolution/Systems/CurvedScalarWave/Test_UpwindFlux.cpp:189]: Please rename this file `Test_UpwindFlux.cpp` -> `Test_UpwindPenaltyCorrection.cpp`

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: tests/Unit/Evolution/Systems/CurvedScalarWave/Test_DuDt.cpp
Comment (prayush) [reply tests/Unit/Evolution/Systems/CurvedScalarWave/Test_DuDt.cpp:96]: done

File: tests/Unit/Evolution/Systems/CurvedScalarWave/Test_UpwindFlux.cpp
Comment (prayush) [reply tests/Unit/Evolution/Systems/CurvedScalarWave/Test_UpwindFlux.cpp:189]: done

File: src/Evolution/Systems/CurvedScalarWave/TimeDerivative.hpp
Comment (prayush) [reply src/Evolution/Systems/CurvedScalarWave/TimeDerivative.hpp]: removed

**Review by nilsdeppe — COMMENTED** [review]
LGTM, please go ahead and squash once @nilsleiffischer has had a chance to look at the fixup

**Comment by prayush** [comment]
> LGTM, please go ahead and squash once @nilsleiffischer has had a chance to look at the fixup

ping @nilsleiffischer

File: src/Evolution/Systems/CurvedScalarWave/TimeDerivative.cpp
Comment (nikwit) [inline src/Evolution/Systems/CurvedScalarWave/TimeDerivative.cpp]: you can delete this forward-declaration as you have already imported `Tensor.hpp` already. You can also delete the `IWYU pragma: keep` comments

File: src/Evolution/Systems/CurvedScalarWave/TimeDerivative.cpp
Comment (nikwit) [inline src/Evolution/Systems/CurvedScalarWave/TimeDerivative.cpp:73]: @nilsdeppe does this system need a `VolumeTermsInstantiation.cpp` like the ScalarWave system has or should this be done later?

File: src/Evolution/Systems/CurvedScalarWave/TimeDerivative.hpp
Comment (nikwit) [inline src/Evolution/Systems/CurvedScalarWave/TimeDerivative.hpp]: delete `Protocols.hpp`, `ProtocolHelpers.hpp`, `FaceNormal.hpp`,  `Options.hpp`, and `Magnitude.hpp`

File: src/Evolution/Systems/CurvedScalarWave/UpwindPenaltyCorrection.cpp
Comment (nikwit) [inline src/Evolution/Systems/CurvedScalarWave/UpwindPenaltyCorrection.cpp]: can be deleted

File: src/Evolution/Systems/CurvedScalarWave/UpwindPenaltyCorrection.cpp
Comment (nikwit) [inline src/Evolution/Systems/CurvedScalarWave/UpwindPenaltyCorrection.cpp]: remove the reference to rvalue

**Review by nikwit — CHANGES_REQUESTED** [review]
group code review with @nilsleiffischer

File: src/Evolution/Systems/CurvedScalarWave/TimeDerivative.cpp
Comment (nilsdeppe) [reply src/Evolution/Systems/CurvedScalarWave/TimeDerivative.cpp:73]: Either that or the executable needs to include the `VolumeTermsImpl.tpp`. The instantiation is preferred. I wouldn't worry about this right now unless this PR is supposed to get the executable running. Assuming we have an input file test it for the system, CI would have a link failure if the instantiations aren't made

Comment (nilsvu) [reply src/Evolution/Systems/CurvedScalarWave/TimeDerivative.cpp:73]: can the explicit instantiation code be deleted then? Sounds like it won't be needed anymore, and all that needs to be done is add the `VolumeTermsInstantiation.cpp` (either in this PR or in a follow-up)

Comment (nilsdeppe) [reply src/Evolution/Systems/CurvedScalarWave/TimeDerivative.cpp:73]: This file is ~designed~ written the way it should be, please leave it be

Comment (nilsvu) [reply src/Evolution/Systems/CurvedScalarWave/TimeDerivative.cpp:73]: Fine with me to leave as-is for now, I don't understand your comment though. If I understand correctly this explicit instantiation code will be deleted and replaced with a `VolumeTermsInstantiation.cpp` to get the executable running.

Comment (nilsdeppe) [reply src/Evolution/Systems/CurvedScalarWave/TimeDerivative.cpp:73]: No, it will not be :)

File: src/Evolution/Systems/CurvedScalarWave/TimeDerivative.cpp
Comment (prayush) [reply src/Evolution/Systems/CurvedScalarWave/TimeDerivative.cpp]: done

File: src/Evolution/Systems/CurvedScalarWave/TimeDerivative.hpp
Comment (prayush) [reply src/Evolution/Systems/CurvedScalarWave/TimeDerivative.hpp]: removed

File: src/Evolution/Systems/CurvedScalarWave/UpwindPenaltyCorrection.cpp
Comment (prayush) [reply src/Evolution/Systems/CurvedScalarWave/UpwindPenaltyCorrection.cpp]: done

File: src/Evolution/Systems/CurvedScalarWave/UpwindPenaltyCorrection.cpp
Comment (prayush) [reply src/Evolution/Systems/CurvedScalarWave/UpwindPenaltyCorrection.cpp]: removed

**Review by nikwit — DISMISSED** [review]
this looks good to me now

**Review by nilsvu — DISMISSED** [review]
LGTM, please squash once @nilsdeppe has had a chance to look. Also, please rebase on latest develop while squashing.

No need to wait for my re-approval after this gets squashed.

**Review by nilsdeppe — COMMENTED** [review]
looks good

**Review by nilsvu — APPROVED** [review]
@nilsdeppe this got squashed, could you take another look?

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**
